### PR TITLE
[GEN] Add cache control to 2D block load/store

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

File: third_party/intel/include/Dialect/TritonGEN/IR/TritonGENAttrDefs.td [hunk 1]
```diff
@@ -54,21 +54,34 @@ def TritonGEN_PrecisionTypeAttr : I32EnumAttr<"PrecisionType",
...
 }
 
-/// Enum attribute of the different cache control settings for a prefetch operation.
-def TritonGEN_PrefetchCacheControl : I32EnumAttr<"PrefetchCacheControl",
-  "TritonGEN prefetch cache control",
+/// Enum attribute of the different cache control settings for a load operation.
+def TritonGEN_LoadCacheControl : I32EnumAttr<"LoadCacheControl",
+  "TritonGEN load cache control",
   [
     I32EnumAttrCase<"DEFAULT",     0,  "Default">,
```
Comment (chengjunlu) [inline third_party/intel/include/Dialect/TritonGEN/IR/TritonGENAttrDefs.td:61]: What is the equivalent semantic of default?

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp [hunk 2]
```diff
@@ -294,7 +294,9 @@ struct LoadOpConversion
             getIntAttr(i1_ty,
                        (isOperandA || eltTy.getIntOrFloatBitWidth() == 32)
                            ? /*A vnni=false*/ 0
-                           : /*B vnni=true*/ 1));
+                           : /*B vnni=true*/ 1),
+            TritonGEN::LoadCacheControlAttr::get(
```
Comment (chengjunlu) [inline third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:298]: Can we create a simple builder that use the `DEFAULT` cache control attribute to build the Gen Op.

Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:298]: Do you mean making cache control operand as optional?

Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp]: done.

**Review by Dewei-Wang-sh — APPROVED** [review]
lgtm

**Review by chengjunlu — APPROVED** [review]

**Comment by chengjunlu** [comment]
@victor-eds 
Thanks for the code changes. It looks good to me.
I'd like to pick it and make it a formal code PR.

**Comment by whitneywhtsang** [comment]
@chengjunlu is this message https://github.com/intel/intel-xpu-backend-for-triton/pull/1131#issuecomment-2111654539 supposed to be posted in a different pr?

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp [hunk 3]
```diff
@@ -280,21 +280,15 @@ struct LoadOpConversion
...
+            /*tile_width*/ elemsPerInstr[1],
+            /*tile_height*/ elemsPerInstr[0],
+            /*v_blocks*/ 1,
+            /*transpose*/ false,
             /*vnni_transform*/
-            getIntAttr(i1_ty,
-                       (isOperandA || eltTy.getIntOrFloatBitWidth() == 32)
-                           ? /*A vnni=false*/ 0
-                           : /*B vnni=true*/ 1));
+            (isOperandA || eltTy.getIntOrFloatBitWidth() == 32)
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:289]: Just pass: ` !isOperandA && eltTy.getIntOrFloatBitWidth() != 32`

Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:289]: Addressed in https://github.com/intel/intel-xpu-backend-for-triton/pull/1158.

**Pull request merged**